### ISSUE TITLE
issues creating phyloseq object in dada2 vis script

### ISSUE THREAD

**Issue by sydneygreenlee** [BigelowLab/edna-dada2#36]
Just adding in case someone else runs into the same issue, feel free to remove if not helpful.

**TLDR: Delete "E-" appended to beginnings of raw sequence file names as a quick fix for sequence names not matching for phyloseq .**

When building my phyloseq object, I got the error:

**Error in validObject(.Object) : invalid class “phyloseq” object: Component sample names do not match. Try sample_names()**

However, in text editors/excel the sample names (e.g. "E-HAB0387_S193_L001_R1_001.fastq.gz") were the same in the relevant files.

Running sample_names(taxa) and sample_names(metadata) each returned NULL

My ASV_taxa.csv had 5310 ASVs.
My seqtab-nochim.csv had 5310 ASVs across 153 samples.
My metadata file had 153 samples with the exact same filenames as my seqtab-nochim.csv in the same format as the tutorial example.

The "E-" appended to each filename was changed to "E." during postprocessing or during import to R, which means the sequence names wouldn't match for phyloseq.
By deleting the "E-" or "E." that was appended to each filename in 'seqtab-nochim.csv' and my metadata .csv, this issue was resolved.

**Comment by btupper** [comment]
The reading function should permit the ingest of names starting with `E-` without converting to `E.`  But we would have to see where that corruption is happening.

I think @robinsleith designed the sample name parsing to be [ruled based](https://github.com/BigelowLab/dadautils/blob/0c20ba92bafb1d157074db3676439581bee3f7af/R/misc.R#L125) I think we may be able to expand the set of rules to deal with `E-`

**Comment by robinsleith** [comment]
This is an issue reading in the pipeline outputs with read.csv in base R, which considers - as an invalid character. If you run check.names = FALSE the - are allowed. I will test that this works downstream in the script and then consider adding check.names = FALSE, since many sequencing centers include - in their filenames.
